### PR TITLE
test: Add more orgs tests

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?
Adds tests to verify
- Org creation and onboarding
- Org member invitation

Based on @leog's work in https://github.com/calcom/cal.com/pull/11804 as I didn't have write access to the PR. I did some small fixes and code cleanup.

## How should this be tested?
`yarn e2e organization`

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: apps/web/pages/signup.tsx [hunk 1]
```diff
@@ -143,7 +143,7 @@ export default function Signup({ prepopulateFormValues, token, orgSlug, orgAutoA
                     methods.clearErrors("apiError");
                   }
 
-                  if (methods.getValues().username === undefined && isOrgInviteByLink && orgAutoAcceptEmail) {
+                  if (!methods.getValues().username && isOrgInviteByLink && orgAutoAcceptEmail) {
```
Comment (hariombalhara) [inline apps/web/pages/signup.tsx:146]: empty username isn't valid so, let's fix that too.

File: apps/web/pages/signup.tsx [hunk 2]
```diff
@@ -37,7 +37,7 @@ type SignupProps = inferSSRProps<typeof getServerSideProps>;
 const checkValidEmail = (email: string) => z.string().email().safeParse(email).success;
 
 const getOrgUsernameFromEmail = (email: string, autoAcceptEmailDomain: string) => {
-  const [emailUser, emailDomain] = email.split("@");
+  const [emailUser, emailDomain = ""] = email.split("@");
```
Comment (hariombalhara) [inline apps/web/pages/signup.tsx:40]: The issue arises when form is submitted without entering the email or email doesn't have @ in it when it was submitted.

File: packages/trpc/server/routers/viewer/organizations/create.handler.ts [hunk 3]
```diff
@@ -175,7 +170,6 @@ export const createHandler = async ({ input, ctx }: CreateOptions) => {
 
     return { user: { ...createOwnerOrg, password } };
   } else {
-    if (!IS_PRODUCTION) return { checked: true };
```
Comment (hariombalhara) [inline packages/trpc/server/routers/viewer/organizations/create.handler.ts:178]: I don't think it's needed. Code verification still accepts any code for dev mode, so there is no need to skip sending email

**Review by hariombalhara — COMMENTED** [review]
Self reviewed

**Comment by hariombalhara** [comment]
Current dependencies on/for this PR:
* main
  * **PR #12241** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/12241?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/calcom/cal.com/12241?utm_source=stack-comment).

**Review by keithwillcode — APPROVED** [review]
Amazing work on these tests, @hariombalhara

**Pull request merged**